drivers/input/input_stmpe811.c

```diff
@@ -0,0 +1,481 @@
+/**
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT st_stmpe811
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/input/input.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(stmpe811, CONFIG_INPUT_LOG_LEVEL);
+
+#include "input_stmpe811.h"
+
+struct stmpe811_config {
+	struct i2c_dt_spec bus;
+	struct gpio_dt_spec int_gpio;
+	uint16_t screen_width;
+	uint16_t screen_height;
+	uint8_t panel_driver_settling_time_us;
+	uint8_t touch_detect_delay_us;
+	uint8_t touch_average_control;
+	uint16_t hold_timeout_ms;
+	uint16_t polling_period_ms;
+	uint16_t release_counter_threshold;
+};
+
+struct stmpe811_data {
+	const struct device *dev;
+	struct k_work work;
```

Comment:
may be a good idea to rename this one since you have another work in the structure